### PR TITLE
Move TCL test integration/slotimport to Go case

### LINKED ISSUE #946: Move TCL test integration/slotimport to Go case

### PR DESCRIPTION
**Pull request by tisonkun**
This closes #946.

### CODE REVIEW

File: tests/gocase/integration/slotimport/slotimport_test.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+		slotNum := 10
+		slotKey := util.GetKeyWithSlotNum(slotNum)
+
+		require.Equal(t, "OK", rdb.Do(ctx, "cluster", "import", slotNum, 0).Val())
+		require.NoError(t, rdb.Set(ctx, slotKey, "slot10_again", 0).Err())
+		require.Equal(t, "slot10_again", rdb.Get(ctx, slotKey).Val())
+
+		// import error
+		require.Equal(t, "OK", rdb.Do(ctx, "cluster", "import", slotNum, 2).Val())
+		time.Sleep(50 * time.Millisecond)
```
Comment (tisonkun) [inline tests/gocase/integration/slotimport/slotimport_test.go:115]: I guess these sleeps will generate unstable tests. We can keep an eye on these cases later.

File: tests/gocase/util/slot.go [hunk 2]
```diff
@@ -0,0 +1,852 @@
...
+func GetKeyWithSlotNum(slotNum int) string {
+	return slotTable[slotNum]
+}
+
+// A table of the shortest possible alphanumeric string that is mapped by
+// redis's crc16 to any given redis cluster slot.
+//
+// The array indexes are slot numbers, so that given a desired slot, this string is guaranteed
+// to make redis cluster route a request to the shard holding this slot
+var slotTable = [...]string{
```
Comment (PragmaTwice) [inline tests/gocase/util/slot.go:31]: Could some comments be added to explain how this table was generated?

Comment (tisonkun) [reply tests/gocase/util/slot.go:31]: I don't know how it's generated also. Ditto the original commit on Kvrocks or Redis: https://github.com/redis/redis/blob/d96f47cf06b1cc24b82109e0e87ac5428517525a/src/crc16_slottable.h

I guess it's random trying and filter..

Comment (PragmaTwice) [reply tests/gocase/util/slot.go:31]: 🤣

**Comment by tisonkun** [comment]
cc @git-hulk @ShooterIT @ChrisZMF please help with reviewing.

**Review by git-hulk — DISMISSED** [review]

File: src/cluster.cc [hunk 3]
```diff
@@ -531,7 +531,7 @@ Status Cluster::ParseClusterNodes(const std::string &nodes_str, ClusterNodes *no
     std::string host = fields[1];
 
     // 3) port
-    auto parse_result = ParseInt<int>(fields[2], NumericRange<int>{1, 65535 - kClusterPortIncr - 1}, 10);
+    auto parse_result = ParseInt<int>(fields[2], 10);
```
Comment (PragmaTwice) [inline src/cluster.cc]: I think `ParseInt<uint16_t>` is more concrete here, since the runtime range is removed.

Comment (tisonkun) [reply src/cluster.cc]:
```suggestion
    auto parse_result = ParseInt<uint16_t>(fields[2], 10);
```

**Review by git-hulk — APPROVED** [review]

**Pull request merged**